### PR TITLE
SLI-92 NPE when binding a project

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/config/global/SonarQubeServerMgmtPanel.java [hunk 1]
```diff
@@ -292,16 +298,20 @@ private void switchTo(@Nullable SonarQubeServer server) {
...
-
+    ConnectedSonarLintEngine.State currentState = engine.getState();
     StringBuilder builder = new StringBuilder();
-    switch (state) {
+
+    switch (currentState) {
       case NEVER_UPDATED:
         builder.append("never updated");
         break;
       case UPDATED:
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/config/global/SonarQubeServerMgmtPanel.java]: MAJOR Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1151)

File: src/main/java/org/sonarlint/intellij/trigger/FileEditorTrigger.java [hunk 2]
```diff
@@ -35,6 +35,7 @@
 import org.sonarlint.intellij.analysis.SonarLintAnalyzer;
 import org.sonarlint.intellij.config.global.SonarLintGlobalSettings;
 import org.sonarlint.intellij.issue.IssueStore;
+import org.sonarlint.intellij.ui.SonarLintConsole;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/trigger/FileEditorTrigger.java]: MINOR Remove this unused import 'org.sonarlint.intellij.ui.SonarLintConsole'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: src/main/java/org/sonarlint/intellij/config/global/SonarQubeServerMgmtPanel.java [hunk 3]
```diff
@@ -292,16 +298,20 @@ private void switchTo(@Nullable SonarQubeServer server) {
...
-        GlobalUpdateStatus updateStatus = engine.getUpdateStatus();
-        builder.append(SonarLintUtils.age(updateStatus.getLastUpdateDate().getTime()));
+        GlobalUpdateStatus updateStatus =  engine.getUpdateStatus();
+        if(updateStatus != null) {
```
Comment (henryju) [inline src/main/java/org/sonarlint/intellij/config/global/SonarQubeServerMgmtPanel.java]: Code style

**Comment by henryju** [comment]
Code style issue then LGTM

**Pull request merged**